From a19b8fff6f84ecf1199be3b0a97325a0cab437c5 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <grinberg@compulab.co.il>
Date: Mon, 23 Mar 2015 09:59:54 +0200
Subject: [PATCH 48/59] ARM: i.MX6: dts: refactor the sbc-fx6 target files

In preparation for DL/S support addition, we move the common code
to dtsi files for better reuse of the same DT code.
Refactor the sbc-fx6 target files.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
[grinberg@compulab.co.il: added a bit more descriptive commit message]
Signed-off-by: Igor Grinberg <grinberg@compulab.co.il>
---
 arch/arm/boot/dts/imx6q-sbc-fx6.dts    |   42 ++--------------------------
 arch/arm/boot/dts/imx6q-sbc-fx6m.dts   |   47 --------------------------------
 arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi  |   37 +++++++++++++++++++++++++
 arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi |   47 ++++++++++++++++++++++++++++++++
 4 files changed, 86 insertions(+), 87 deletions(-)

diff --git a/arch/arm/boot/dts/imx6q-sbc-fx6.dts b/arch/arm/boot/dts/imx6q-sbc-fx6.dts
index 9d31d15..1234fb3 100644
--- a/arch/arm/boot/dts/imx6q-sbc-fx6.dts
+++ b/arch/arm/boot/dts/imx6q-sbc-fx6.dts
@@ -18,44 +18,6 @@
 #include "imx6qdl-sb-fx6.dtsi"
 
 / {
-   model = "CompuLab CM-FX6 on SBC-FX6";
-   compatible = "compulab,cm-fx6", "compulab,sbc-fx6", "fsl,imx6q";
-
-	backlight {
-		compatible = "pwm-backlight";
-		pwms = <&pwm3 0 5000000>;
-		brightness-levels = <0 4 8 16 32 64 128 255>;
-		default-brightness-level = <7>;
-	};
-
-};
-
-&i2c1 {
-	status = "okay";
-};
-
-&usdhc3 {
-	wp-gpios = <&gpio7 0 0>;
-	status = "okay";
-};
-
-&pwm3 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm3_1>;
-	status = "okay";
-};
-
-&mxcfb1 {
-	status = "okay";
-};
-
-&mxcfb2 {
-	mode_str ="KD050C-WVGA";
-	status = "okay";
-};
-
-&flexcan1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_flexcan1_1>;
-	status = "okay";
+	model = "CompuLab CM-FX6 on SBC-FX6";
+	compatible = "compulab,cm-fx6", "compulab,sbc-fx6", "fsl,imx6q";
 };
diff --git a/arch/arm/boot/dts/imx6q-sbc-fx6m.dts b/arch/arm/boot/dts/imx6q-sbc-fx6m.dts
index a98f1a2..19bf948 100644
--- a/arch/arm/boot/dts/imx6q-sbc-fx6m.dts
+++ b/arch/arm/boot/dts/imx6q-sbc-fx6m.dts
@@ -20,51 +20,4 @@
 / {
 	model = "CompuLab CM-FX6 on SBC-FX6m";	
 	compatible = "compulab,cm-fx6", "compulab,sbc-fx6m", "fsl,imx6q";
-
-	iomux_uart2: pinmux@20E0924 {
-		compatible = "pinctrl-single";
-		reg = <0x20E0000 0x924>;	/* Single register */
-		#address-cells = <1>;
-		#size-cells = <0>;
-		pinctrl-single,register-width = <32>;
-		pinctrl-single,function-mask = <0x4>;
-	};
-};
-
-&iomuxc {
-	imx6q-sbc-fx6m {
-		/* pins for uart2 */
-		pinctrl_uart2: uart2grp {
-			fsl,pins = <
-				MX6QDL_PAD_GPIO_7__UART2_TX_DATA 0x1b0b1
-				MX6QDL_PAD_GPIO_8__UART2_RX_DATA 0x1b0b1
-				MX6QDL_PAD_SD4_DAT5__UART2_RTS_B 0x1b0b1
-				MX6QDL_PAD_SD4_DAT6__UART2_CTS_B 0x1b0b1
-		>;
-		};
-	};
-};
-
-&i2c1 {
-	status = "okay";
-};
-
-&usdhc3 {
-	status = "okay";
-};
-
-/* rear serial console */
-&uart2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart2>;
-	fsl,uart-has-rtscts;
-	status = "okay";
-};
-
-&mxcfb1 {
-	status = "okay";
-};
-
-&mxcfb2 {
-	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi b/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
index 4d030f9..129e88e 100644
--- a/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
@@ -12,6 +12,13 @@
  */
 
 / {
+	backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm3 0 5000000>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <7>;
+	};
+
 	i2cmux {
 		compatible = "i2c-mux-gpio";
 		#address-cells = <1>;
@@ -58,3 +65,33 @@
 
 	};
 };
+
+&i2c1 {
+	status = "okay";
+};
+
+&usdhc3 {
+	wp-gpios = <&gpio7 0 0>;
+	status = "okay";
+};
+
+&pwm3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm3_1>;
+	status = "okay";
+};
+
+&mxcfb1 {
+	status = "okay";
+};
+
+&mxcfb2 {
+	mode_str ="KD050C-WVGA";
+	status = "okay";
+};
+
+&flexcan1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan1_1>;
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi b/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi
index 5e6c859..5394364 100644
--- a/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi
@@ -12,6 +12,15 @@
  */
 
 / {
+	iomux_uart2: pinmux@20E0924 {
+		compatible = "pinctrl-single";
+		reg = <0x20E0000 0x924>;	/* Single register */
+		#address-cells = <1>;
+		#size-cells = <0>;
+		pinctrl-single,register-width = <32>;
+		pinctrl-single,function-mask = <0x4>;
+	};
+
 	eth@pcie {
 		compatible = "intel,i211";
 		local-mac-address = [FF FF FF FF FF FF];
@@ -64,3 +73,41 @@
 		};
 	};
 };
+
+&iomuxc {
+	imx6q-sbc-fx6m {
+		/* pins for uart2 */
+		pinctrl_uart2: uart2grp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_7__UART2_TX_DATA 0x1b0b1
+				MX6QDL_PAD_GPIO_8__UART2_RX_DATA 0x1b0b1
+				MX6QDL_PAD_SD4_DAT5__UART2_RTS_B 0x1b0b1
+				MX6QDL_PAD_SD4_DAT6__UART2_CTS_B 0x1b0b1
+		>;
+		};
+	};
+};
+
+&i2c1 {
+	status = "okay";
+};
+
+&usdhc3 {
+	status = "okay";
+};
+
+/* rear serial console */
+&uart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2>;
+	fsl,uart-has-rtscts;
+	status = "okay";
+};
+
+&mxcfb1 {
+	status = "okay";
+};
+
+&mxcfb2 {
+	status = "okay";
+};
-- 
1.7.9.5

